### DISCUSSION TITLE
Null handling

### DISCUSSION THREAD

**Discussion by andresmarenco-pureinsights** [ezylang/EvalEx#395] [Q&A]
Hi,

I'm using version 3.0.5 of EvalEx and I'm facing the issue of unsupported nulls. I see this will be solved for the next release.

Is there an estimated release date for 3.1.0? Can I use 3.1.0-SNAPSHOT in the meantime without worrying about breaking changes?

Or any chance to have a 3.0.6 release with only that feature?

If not, is there any workaround?

**Comment by uklimaschewski** [comment]
Hi, I planned to have it released already. But I have very limited time at the moment.
I hope to find a few free days in the next one or two months to fix some issues with the date/time functionality.
The snapshot will have some small breaking changes, but mainly for the new date/time support.
Also, there won't be a 3.0.6 patch release with new functionality. I follow semantic versioning, so any new functionality will go into a new minor release.

**Reply by andresmarenco-pureinsights** [reply]
That sounds great! Thanks!

Btw, I was planning to write a `coalesce()` function, similar to the one from SQL where it accepts any number of parameters and returns the first non-null value. Would that work for you? I can send you a PR if that helps.

**Reply by uklimaschewski** [reply]
I have just merged some bigger changes for 3.1.0, I added a configurable type conversion, instead of doing all this stuff in the EvaluationValue. The old methods are still there, but deprecated.
This was a major step in the path to the the 3.1.0 release.
Feel free to create a PR, based on the new main branch. Such a coalesce function makes sense in combination with the new null support.

**Reply by omniavincitlabor** [reply]
Are there plans to make operators null-aware?
for example: null + 1 = 1; null - 1 = -1 ; null * 12 = null ...

**Reply by uklimaschewski** [reply]
There are no plans yet. I think there is no common definition of how nulls are handled in general.
This highly depends on the operator or function. And also on personal opinion, I'm afraid.
